supertokens_python/recipe/session/recipe_implementation.py

```diff
@@ -152,24 +168,118 @@ async def create_new_session(
         session_data: Union[None, Dict[str, Any]],
         user_context: Dict[str, Any],
     ) -> SessionContainer:
-        session = await session_functions.create_new_session(
-            self, user_id, access_token_payload, session_data
+        log_debug_message("createNewSession: Started")
+        output_transfer_method = self.config.get_token_transfer_method(
+            request, True, user_context
         )
-        access_token = session["accessToken"]
-        refresh_token = session["refreshToken"]
-        id_refresh_token = session["idRefreshToken"]
+        if output_transfer_method == "any":
+            output_transfer_method = "header"
+
+        log_debug_message(
+            "createNewSession: using transfer method %s", output_transfer_method
+        )
+
+        if (
+            (output_transfer_method == "cookie")
+            and self.config.cookie_same_site == "none"
+            and not self.config.cookie_secure
+            and not (
+                (
+                    self.app_info.top_level_api_domain == "localhost"
+                    or is_an_ip_address(self.app_info.top_level_api_domain)
+                )
+                and (
+                    self.app_info.top_level_website_domain == "localhost"
+                    or is_an_ip_address(self.app_info.top_level_website_domain)
+                )
+            )
+        ):
+            # We can allow insecure cookie when both website & API domain are localhost or an IP
+            # When either of them is a different domain, API domain needs to have https and a secure cookie to work
+            raise Exception(
+                "Since your API and website domain are different, for sessions to work, please use "
+                "https on your apiDomain and don't set cookieSecure to false."
+            )
+
+        disable_anti_csrf = output_transfer_method == "header"
+
+        result = await session_functions.create_new_session(
+            self,
+            user_id,
+            disable_anti_csrf,
+            access_token_payload,
+            session_data,
+        )
+
+        response_mutators: List[Callable[[Any], None]] = []
```

Comment:
Please do strong typing for this everywhere in this file

Reply:
Fixed.